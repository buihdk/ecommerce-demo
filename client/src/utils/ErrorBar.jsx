import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Box, Snackbar, SnackbarContent } from '@material-ui/core';
import { Error } from '@material-ui/icons';

const renderMessage = msg => (
  <Box display="flex" alignItems="center">
    <Error />
    <span style={{ marginLeft: 4 }}>{msg}</span>
  </Box>
);

const ErrorBar = ({ msg }) => (
  <Snackbar open anchorOrigin={{ vertical: 'top', horizontal: 'center' }}>
    <SnackbarContent
      style={{ backgroundColor: '#e43' }}
      message={renderMessage(msg)}
    />
  </Snackbar>
);

ErrorBar.propTypes = {
  msg: PropTypes.string,
};

ErrorBar.defaultProps = {
  msg: '',
};

export default memo(ErrorBar);
