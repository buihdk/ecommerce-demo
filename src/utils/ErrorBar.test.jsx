import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ErrorBar from './ErrorBar';

describe('ErrorBar', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<ErrorBar msg="mock message" />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
