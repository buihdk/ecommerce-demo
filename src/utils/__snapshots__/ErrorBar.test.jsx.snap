// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBar renders without crashing 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "center",
      "vertical": "top",
    }
  }
  open={true}
>
  <WithStyles(ForwardRef(SnackbarContent))
    message={
      <ForwardRef(StyledComponent)
        alignItems="center"
        display="flex"
      >
        <UNDEFINED />
        <span
          style={
            Object {
              "marginLeft": 6,
            }
          }
        >
          mock message
        </span>
      </ForwardRef(StyledComponent)>
    }
    style={
      Object {
        "backgroundColor": "#e43",
      }
    }
  />
</WithStyles(ForwardRef(Snackbar))>
`;
